soh/src/code/z_parameter.c

```diff
@@ -1851,9 +1851,17 @@ u8 Item_Give(PlayState* play, u8 item) {
             }
             
         } else if (item == ITEM_SWORD_MASTER) {
-            gSaveContext.equips.buttonItems[0] = ITEM_SWORD_MASTER;
-            gSaveContext.equips.equipment &= 0xFFF0;
-            gSaveContext.equips.equipment |= 0x0002;
+            /*Forces the sword to the Master Sword when:
+            - Drawing the Master Sword from its pedestal
+            - Fighting Ganon, after the first phase
+            Breaks DList Rework for drawing the sword, needed for Gannon Fight.
+            To fix this, only run this code as when Link is an adult, meaning Link MUST be fighting Ganon*/
```

Comment:
alternate loading detects to see if the proper model can use the new system.

Reply:
If it's preferred it could be changed like this:
```diff
-if (LINK_IS_ADULT)
+if (!ResourceGetIsCustomByName(gLinkMasterSwordDL) || LINK_IS_ADULT)
```
This way the code would be executed only if link is adult when there's a custom master sword dl and it would always be executed when there isn't one.